apis/Google.Cloud.Bigtable.V2/Google.Cloud.Bigtable.V2/RowFilters.cs

```diff
@@ -0,0 +1,380 @@
+        /// representations will be used.
+        /// </para>
+        /// </remarks>
+        /// <param name="regex">
+        /// The RE2 regex used to match column qualifiers.
+        /// </param>
+        /// <returns>The created row filter.</returns>
+        public static RowFilter ColumnQualifierRegex(BigtableByteString regex) =>
+            new RowFilter { ColumnQualifierRegexFilter = regex.Value };
+
+        /// <summary>
+        /// Creates a new <see cref="RowFilter"/> instance which matches only cells from
+        /// columns within the given range.
+        /// </summary>
+        /// <param name="range">
+        /// The range of columns from which cells should be matched. Must not be null.
+        /// </param>
+        /// <returns>The created row filter.</returns>
+        public static RowFilter ColumnRange(ColumnRange range) =>
+            new RowFilter { ColumnRangeFilter = GaxPreconditions.CheckNotNull(range, nameof(range)) };
+
+        // TODO: Add additional ColumnRange... methods to take start/end values?
+
+        /// <summary>
+        /// Creates a new <see cref="RowFilter"/> instance which applies one of two
+        /// possible RowFilters to the data based on the output of a predicate
+        /// RowFilter.
+        /// </summary>
+        /// <param name="predicateFilter">
+        /// If <paramref name="predicateFilter"/> outputs any cells, then
+        /// <paramref name="trueFilter"/> will be evaluated on the input row.
+        /// Otherwise, <paramref name="falseFilter"/> will be evaluated.
+        /// Must not be null.
+        /// </param>
+        /// <param name="trueFilter">
+        /// The filter to apply to the input row if <paramref name="predicateFilter"/>
+        /// returns any results. If null, no results will be returned in the true case.
+        /// </param>
+        /// <param name="falseFilter">
+        /// The filter to apply to the input row if <paramref name="predicateFilter"/>
+        /// does not return any results. If null, no results will be returned in the
+        /// false case.
+        /// </param>
+        /// <returns>The created row filter.</returns>
+        public static RowFilter Condition(
+            RowFilter predicateFilter, RowFilter trueFilter, RowFilter falseFilter) =>
+            new RowFilter
+            {
+                Condition = new RowFilter.Types.Condition
+                {
+                    // TODO: Verify this on server. Might be allowed for (row contains any cells).
+                    PredicateFilter = GaxPreconditions.CheckNotNull(predicateFilter, nameof(predicateFilter)),
+                    TrueFilter = trueFilter,
+                    FalseFilter = falseFilter
+                }
+            };
+
+        /// <summary>
+        /// Creates a new <see cref="RowFilter"/> instance which matches only cells
+        /// from columns whose families satisfy the given RE2 regex.
```

Comment:
Is "RE2" a well-understood term within Bigtable? Presumably it's as per https://github.com/google/re2

Reply:
I'm assuming that's it, but I'll confirm and maybe add links in the remarks section.